exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/TraceExporter.java

```diff
@@ -1,25 +1,79 @@
 package com.google.cloud.opentelemetry.trace;
 
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
 import com.google.cloud.trace.v2.TraceServiceClient;
+import com.google.cloud.trace.v2.TraceServiceSettings;
+import com.google.cloud.trace.v2.stub.TraceServiceStub;
 import com.google.devtools.cloudtrace.v2.AttributeValue;
 import com.google.devtools.cloudtrace.v2.ProjectName;
 import com.google.devtools.cloudtrace.v2.Span;
 import io.opentelemetry.sdk.trace.data.SpanData;
 import io.opentelemetry.sdk.trace.export.SpanExporter;
+
+import java.io.IOException;
+import java.time.Duration;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
 import java.util.Map;
 
+import static com.google.api.client.util.Preconditions.checkNotNull;
+
 public class TraceExporter implements SpanExporter {
 
   private final TraceServiceClient traceServiceClient;
   private final ProjectName projectName;
   private final String projectId;
   private final Map<String, AttributeValue> fixedAttributes;
 
-  public TraceExporter(String projectId, TraceServiceClient traceServiceClient,
-      Map<String, AttributeValue> fixedAttributes) {
+  static TraceExporter createWithConfiguration(TraceConfiguration configuration)
+          throws IOException {
+    String projectId = configuration.getProjectId();
+    TraceServiceStub stub = configuration.getTraceServiceStub();
+    Credentials credentials = configuration.getCredentials();
+
+    if (stub == null) {
+      return TraceExporter.createWithCredentials(
+              projectId,
+              credentials != null ? credentials : GoogleCredentials.getApplicationDefault(),
+              configuration.getFixedAttributes(),
+              configuration.getDeadline());
+    }
+    return TraceExporter.createWithStub(
+            projectId, TraceServiceClient.create(stub), configuration.getFixedAttributes());
+  }
+
+  static TraceExporter createWithStub(
```

Comment:
Stub refers to testing, and this method doesn't actually explicitly use stubs internally, so I suggest to rename it to

```suggestion
  static TraceExporter createWithClient(
```